prjxray/verilog.py

```diff
@@ -65,6 +66,37 @@ def unquote(s):
     return s[1:-1]
 
 
+def to_int(s):
+    value = 0
+
+    match = re.search(r'^(\d+)\'([sS]*)([bBoOdDhH])(.*)', str(s))
+
+    if match:
+        width = int(match.group(1))
+        signed = match.group(2)
+        radix = match.group(3)
+        value = match.group(4)
+
+        # Convert to int type
+        if re.match(r'[bB]', radix):
+            value = int(value, 2)
+        elif re.match(r'[oO]', radix):
+            value = int(value, 8)
+        elif re.match(r'[dD]', radix):
+            value = int(value, 10)
+        elif re.match(r'[hH]', radix):
+            value = int(value, 16)
+        else:
+            raise ValueError('Don\'t know how to interpret input %s' % (s))
+
+        # Truncate to `width` bits
+        value &= 2**width - 1
```

Comment:
This should probably be an assertion, because I don't believe it is valid verify to provide a value higher than the width specifier.

Reply:
I can't find anywhere in the [IEEE Verilog 2001 Standard](http://www-inst.eecs.berkeley.edu/~cs150/fa06/Labs/verilog-ieee.pdf) that mentions whether a higher value is illegal or is truncated.  [This source](http://web.engr.oregonstate.edu/~traylor/ece474/beamer_lectures/verilog_number_literals.pdf) says that the value is truncated and zero-filled.  Naturally I would assume truncation, e.g. if you had a large, odd number of bits and wanted to fill out the Msb using hexadecimal notation like `31'hFFFFFFFF`.

Reply:
Fair enough, it doesn't seem pretty weird to allow it.